draft-ietf-jsonpath-base.md

```diff
@@ -287,23 +287,22 @@ This section is informative.
 
 This document picks up {{{Stefan Gössner}}}'s popular JSONPath proposal
 dated 2007-02-21 {{JSONPath-orig}}, builds on the experience from the widespread
-deployment of its implementations, and provides a normative definition for it.
+deployment of its implementations, and provides a normative specification for it.
 
 {{inspired-by-xpath}} describes how JSONPath was inspired by XML's XPath
 [XPath].
 
 JSONPath was intended as a light-weight companion to JSON
-implementations on platforms such as PHP and JavaScript, so instead of
-defining its own expression language like XPath did, JSONPath
-delegated this to the expression language of the platform.
-While the languages in which JSONPath is used do have significant
-commonalities, over time this caused non-portability of JSONPath
-expressions between the ensuing platform-specific dialects.
-
-The present specification aims to remove platform dependencies and
+implementations in languages such as PHP and JavaScript, so instead of
+defining its own expression language, like XPath did, JSONPath
+delegated to the implementation language.
+As JSONPath was implemented in more languages, JSONPath
+expressions became decreasingly portable.
+
+This document aims to remove language dependencies and
 serve as a common JSONPath specification that can be used across
-platforms.  Obviously, this means that backwards compatibility could
-not always be achieved; a design principle of this specification is to
+languages.  Obviously, this means that backwards compatibility could
```

Comment:
I have no idea what languages have to do with this.
Platform was the right word.

Reply:
I'm concerned that JSONPath should interoperate between implementations in distinct languages on the same platform (as well as between platforms). Make sense?

Reply:
The implementation language is part of the platform.

(I didn't mean OS platforms.)

The user of the JSONPath implementation should not be concerned with the implementation language.
Multiple platforms can be implemented in the same language (e.g., C or C++).

Reply:
To me, "platform" means either OS or hardware. [Wikipedia](https://en.wikipedia.org/wiki/Computing_platform) expands on that somewhat. So I don't think we can use the unqualified term "platform" and be sure that our readers will understand by it the same thing we do (even if we agreed among ourselves).

Most of my experience is in C, C++, Java, Go, and Rust all of which support multiple OS/hardware platforms. I think I would describe thing you are calling a platform as a language runtime.

Please could you define what you mean by platform?

Reply:
I have a slight preference for "language" also.

Reply:
@cabo PTAL at the latest version of this PR. If you are still uncomfortable with the changes to the history section, please could you define what you mean by platform in this context?

Reply:
The problem is that the variation that we are observing does not come from the programming language only.  The platform (not just OS platform) might provide more variation, e.g., whether RE2 or PCRE or a Boost dialect is available for the RE processing -- all of these could be in a C++ implementation.

If your view of "platform" is more narrowly "OS platform", then we maybe need a term for what I call "platform".